         }
diff -upr atl2-1.4.0.20./src/at_main.c atl2-1.4.0.20/src/at_main.c
--- atl2-1.4.0.20./src/at_main.c	2007-04-21 11:19:25.000000000 +0200
+++ atl2-1.4.0.20/src/at_main.c	2007-08-24 22:49:28.682998750 +0200
@@ -1484,9 +1484,12 @@ at_vlan_rx_kill_vid(struct net_device *n
 //    DEBUGFUNC("at_vlan_rx_kill_vid !");
     at_irq_disable(adapter);
 
-    if (adapter->vlgrp) {
-        adapter->vlgrp->vlan_devices[vid] = NULL;
-    }
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,21)  
+	if (adapter->vlgrp)
+		adapter->vlgrp->vlan_devices[vid] = NULL;
+#else
+	vlan_group_set_device(adapter->vlgrp, vid, NULL);
+#endif
 
     at_irq_enable(adapter);
 
@@ -1504,8 +1507,12 @@ at_restore_vlan(struct at_adapter *adapt
     if(adapter->vlgrp) {
         u16 vid;
         for(vid = 0; vid < VLAN_GROUP_ARRAY_LEN; vid++) {
-            if(!adapter->vlgrp->vlan_devices[vid])
-                continue;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,21)  
+		if(!adapter->vlgrp->vlan_devices[vid])
+#else
+		if(!adapter->vlgrp->vlan_devices_arrays[vid])
+#endif
+			continue;
             at_vlan_rx_add_vid(adapter->netdev, vid);
         }
     }
